spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractReplyProducingMessageHandler.java

```diff
@@ -159,6 +171,18 @@ public String toString() {
 			return AbstractReplyProducingMessageHandler.this.toString();
 		}
 
+		/**
+		 * Utility method, intended for use in message handler advice classes to get
+		 * information about the advised object. For example:
+		 * <p>
+		 * {@code ((AbstractReplyProducingMessageHandler.RequestHandler)
```

Comment:
OTOH why we really can't move `getAdvisedHandler()` to the `RequestHandler` and don't care about `AdvisedRequestHandler` making `public`?